varats-core/varats/report/multi_patch_report.py

```
@@ -0,0 +1,82 @@
+        super().__init__(path)
+        self.__patched_reports: tp.Dict[str, ReportTy] = {}
+
+        with tempfile.TemporaryDirectory() as tmp_result_dir:
+            shutil.unpack_archive(path, extract_dir=tmp_result_dir)
+
+            for report in Path(tmp_result_dir).iterdir():
+                if self.is_baseline_report(report.name):
+                    self.__base = report_type(report)
+                elif self.is_patched_report(report.name):
+                    self.__patched_reports[
+                        self._parse_patch_shorthand_from_report_name(
+                            report.name
+                        )] = report_type(report)
+
+            if not self.__base or not self.__patched_reports:
+                raise AssertionError(
+                    "Reports where missing in the file {report_path=}"
+                )
+
+    def get_baseline_report(self) -> ReportTy:
+        return self.__base
+
+    def get_report_for_patch(self,
+                             patch_shortname: str) -> tp.Optional[ReportTy]:
+        """Get the report for a given patch shortname."""
+        if patch_shortname in self.__patched_reports:
+            return self.__patched_reports[patch_shortname]
+
+        return None
+
+    def get_patch_names(self) -> tp.List[str]:
+        return list(self.__patched_reports.keys())
+
+    def get_patched_reports(self) -> tp.ValuesView[ReportTy]:
+        return self.__patched_reports.values()
+
+    @staticmethod
+    def create_baseline_report_name(base_file_name: str) -> str:
+        return f"baseline_{base_file_name}"
+
+    @staticmethod
+    def is_baseline_report(file_name: str) -> bool:
+        return file_name.startswith("baseline_")
+
+    @staticmethod
+    def create_patched_report_name(patch: Patch, base_file_name: str) -> str:
+        return (
+            f"patched_{len(patch.shortname)}_" +
+            f"{patch.shortname}_{base_file_name}"
+        )
```

Personally I think it's a bit weird to encode the shortname length in the report name, however I can see why it is necessary to extract the patch name if only given a report file name.

A solution to omit the length here might be to just silently convert underscores in the short names to dashes during parsing of the `.info` files, but not sure if that might be confusing for people if their patches are "renamed" (even though if it is just slightly)

---

I would not rename them, not only for confusion reasons but also because then a name with a `-` is problematic. Encoding the length is the only options to allow a free name selection. And it should not really do harm, as user will not see the raw file names anyhow, as they are generated by via the API and after parsing the infos are access only through the API.